### PR TITLE
Implement subscribe api in replication manager

### PR DESCRIPTION
**Pull request by kaustubh-d**
Changes:
- Implement subscribe api in replication manager to register prefetch count
- Add test cases

### CODE REVIEW

File: s3/replication/manager/src/s3replicationmanager/manager_routes.py [hunk 1]
```diff
@@ -33,75 +34,102 @@
...
 @routes.get('/subscribers')  # noqa: E302
 async def list_subscribers(request):
     """Handler to list subscribers."""
-    return web.json_response({'subscribers': json.dumps(
-        list(request.app['subscribers'].get_keys()))}, status=200)
+    _logger.debug('API: GET /subscribers')
+    subscribers = request.app['subscribers']
+
+    _logger.debug('Number of subscribers {}'.format(subscribers.count()))
+    # _logger.debug('List of jobs in-progress {}'.format(Jobs.dumps(jobs)))
```
Comment (rajkumarpatel2602) [inline s3/replication/manager/src/s3replicationmanager/manager_routes.py:59]: we can remove this log from this api.

File: s3/replication/manager/src/s3replicationmanager/manager_routes.py [hunk 2]
```diff
@@ -135,16 +164,18 @@ async def list_jobs(request):
     url_ob = urlparse(url)
     query = parse_qs(url_ob.query)
 
+    _logger.debug('API: GET /jobs\n Query: {}'.format(query))
+
     progressing_jobs_obj = request.app['jobs_in_progress']
     all_jobs_obj = request.app['all_jobs']
 
-    # LOG.debug(
+    # _logger.debug(
```
Comment (rajkumarpatel2602) [inline s3/replication/manager/src/s3replicationmanager/manager_routes.py:172]: these logs can be removed if not required.

File: s3/replication/manager/tests/system/test_subscribers_resource.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+    async with aiohttp.ClientSession() as session:
+        # Add subscriber
+        async with session.post(test_config['url'] + '/subscribers',
+                                json=test_payload) as response:
+            logger.debug('HTTP Response: Status: {}'.format(response.status))
+            if test_case_name == 'valid_payload':
+                response_body = await response.json()
+                logger.debug('HTTP Response: Body: {}'.format(response_body))
+                global global_valid_subscriber_id
```
Comment (rajkumarpatel2602) [inline s3/replication/manager/tests/system/test_subscribers_resource.py:52]: global is not required may be, as we are not modifying value of 'global_valid_subscriber_id'.

Comment (kaustubh-d) [reply s3/replication/manager/tests/system/test_subscribers_resource.py:52]: We are using it in further tests hence global

File: s3/replication/manager/src/s3replicationmanager/manager_routes.py (hunk 2 above)
Comment (kaustubh-d) [reply s3/replication/manager/src/s3replicationmanager/manager_routes.py]: Fixed

File: s3/replication/manager/src/s3replicationmanager/manager_routes.py (hunk 1 above)
Comment (kaustubh-d) [reply s3/replication/manager/src/s3replicationmanager/manager_routes.py]: Fixed

**Review by rajkumarpatel2602 — APPROVED** [review]

**Pull request merged**